File: 프로그래머스/2/42584. 주식가격/주식가격.js
function solution(prices) {
    const stack = [0];
    const answer = new Array(prices.length).fill(0)
    
    for(let i=1; i<prices.length; i+=1){
        while(stack.length>0 && prices[i]<prices[stack.at(-1)]){
            const j = stack.pop()
            answer[j] = i-j;
        }
        stack.push(i)
    }
    
    while(stack.length>0){
        const j = stack.pop()
        answer[j] = prices.length-j-1;
    }
    
    return answer
}

File: 프로그래머스/2/42584. 주식가격/주식가격.py
def solution(prices):
    answer = [0] * len(prices)
    stack = []
    for i in range(len(prices)):
        price = prices[i]
        while True:
            if(stack and prices[stack[-1]]<= price or not stack): break
            s = stack.pop()
            answer[s] = i - s
        stack.append(i)
    
    while stack:
        s = stack.pop()
        answer[s] = len(prices) - s - 1
    return answer